functions/src/monitor/monitor.ts

```diff
@@ -1,48 +1,60 @@
 // 3rd party
+import { firestore } from 'firebase-admin';
+import { v4 as uuid } from 'uuid';
 import { FirebaseAdmin } from '../client/firebase-admin';
 
 // DashboardHub
 import { Ping, PingResponse } from '../client/ping';
-import { Logger } from './../client/logger';
 import { MonitorModel, PingModel, ProjectModel } from './../models/index.model';
 
 type Document = FirebaseFirestore.DocumentSnapshot
+type QuerySnapshot = firestore.QuerySnapshot;
 
 export interface MonitorInfoInput {
   projectUid: string;
   monitorUid: string;
 }
 
-export const getMonitorPings: any = async (projectUid: string, monitorUid: string) => {
+export const ping: any = async (projectUid: string, monitorUid: string) => {
   const document: Document = await FirebaseAdmin.firestore()
     .collection('projects')
     .doc(projectUid)
     .get();
   const project: ProjectModel = <ProjectModel> document.data();
   const monitor: MonitorModel = project.monitors.find((item: MonitorModel) => item.uid === monitorUid);
-  const uri: string = project.url + monitor.path;
+  const url: string = project.url + monitor.path;
+  const uid: string = uuid();
 
   const start: number = (new Date()).getMilliseconds();
-  const response: PingResponse = await Ping<PingResponse>(uri);
-  Logger.info(response);
+  const response: PingResponse = await Ping<PingResponse>(url);
   const end: number = (new Date()).getMilliseconds();
   const pingResult: PingModel = {
+    monitorUid: monitorUid,
     statusCode: response.statusCode,
     expectedCode: monitor.expectedCode,
     expectedText: monitor.expectedText,
-    fullUrl: uri,
-    duration: end - start, // Milliseconds
+    url: url,
+    duration: end - start,
```

Comment:
This is sometimes is negative, because `.getMilliseconds()` is for within the second not the time in milliseconds - I will update to use `Date.now()`